File: task1/src/main/java/ru/biluta/task1/service/TaskManager.java
package ru.biluta.task1.service;

import ru.biluta.task1.model.Task;
import ru.biluta.task1.utils.RandomGenerators;

public class TaskManager {

    private final RandomGenerators generators;

    public TaskManager(double sigma, double lambda, int k) {
        this.generators = new RandomGenerators(sigma, lambda, k);
    }

    public Task createTask() {
        double processingTime = generators.generateRayleigh();
        return new Task(processingTime);
    }

}

File: task2/src/main/kotlin/ru.biluta.task2/TaskManager.kt
package main.kotlin.ru.biluta.task2

class TaskManager(
    private val sigma: Double,
    private val lambda: Double,
    private val k: Int,
    private val generators: RandomGenerators = RandomGenerators(sigma, lambda, k)
) {

    fun createTask(): Task {
        val processingTime = generators.generateRayleigh()
        return Task(processingTime)
    }

}